controllers/vaultdynamicsecret_controller.go

```diff
@@ -84,14 +84,29 @@ func (r *VaultDynamicSecretReconciler) Reconcile(ctx context.Context, req ctrl.R
 		if apierrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
-
 		logger.Error(err, "error getting resource from k8s", "obj", o)
 		return ctrl.Result{}, err
 	}
+	// Add a finalizer on the VDS resource if we intend to Revoke on cleanup path, otherwise
+	// there isn't a need for it since we are not managing anything on deletion.
+	if o.Spec.Revoke {
+		if o.GetDeletionTimestamp() == nil {
+			if err := r.addFinalizer(ctx, o); err != nil {
+				return ctrl.Result{}, err
+			}
+		} else {
+			logger.Info("Got deletion timestamp", "obj", o)
+			if err := r.handleDeletion(ctx, o); err != nil {
+				msg := "Failed to handle deletion"
+				r.Recorder.Eventf(o, corev1.EventTypeWarning, consts.ReasonSecretLeaseRevoked, msg+": %s", err)
+				return ctrl.Result{}, err
+			}
```

Comment:
Since `handleDeletion()` also checks `Spec.Revoke`, I wonder if the else clause should be out on it's own (instead of under the `if o.Spec.Revoke` here).